feat(argocd): enable argocd documentation (#238)
upbound/docs: content/concepts/mcp/control-plane-connector.md

### Argo CD

#### Configure Argo CD

> To configure Argo CD for Annotation resource tracking, edit the argocd-cm

Reviewer: Is `Annotation` supposed to be capitalized? 

Is `argocd-cm` a specific file?

Author: used it from this page: https://docs.crossplane.io/knowledge-base/integrations/argo-cd-crossplane/

ConfigMap in the argocd Namespace. Add `application.resourceTrackingMethod: annotation`
to the data section as below:

```bash
apiVersion: v1

> data:

Reviewer: ```suggestion
```yaml
apiVersion: v1
metadata:
  name: argocd-cm
data:
```
Add in the configmap name to make it obvious what's being edited.

  application.resourceTrackingMethod: annotation
kind: ConfigMap
```

> This will Argo CD not trigger auto-pruning.

Author: not auto-pruning composites, managed resources etc.

> This will Argo CD not trigger auto-pruning.

Reviewer: I'm not familiar enough with ArgoCD to know exactly what to say but maybe "This configuration turns off Argo CD auto pruning, preventing the deletion of Crossplane resources."

> #### Connect Argo CD

Reviewer: ```suggestion
<!-- vale Google.Headings = NO -->
#### Connect Argo CD
<!-- vale Google.Headings = YES -->
```

Can you expand this a bit more? Maybe "Connect an MCP to Argo CD"?

> You can use Argo to sync claims to your managed control planes. To do this, you need to have an instance of Argo running externally to Upbound, since you can't install Argo locally in Upbound. Assuming you are running Argo _elsewhere_ outside of Upbound, the steps to use Argo to sync claims are the following:

Reviewer: ```suggestion
You can connect an external Argo CD instance to your managed control plane to sync Crossplane claims. 
```

> 1. Fetch your managed control plane's API server endpoint so you can provide it to Argo. In the step below, use the [up CLI]({{<ref "concepts/mcp/_index.md#connect-directly-to-your-mcp" >}}) to fetch the kubeconfig of your MCP and write it to a file.

Reviewer: ```suggestion
1. Create a kubeconfig file for your MCP with the [up CLI]({{<ref "concepts/mcp/_index.md#connect-directly-to-your-mcp" >}}). Use the `up ctp kubeconfig get` command and define the filename to save the kubeconfig to. This example saves the kubeconfig to a file named `mcp-kubeconfig.yaml`. 
```
